Symbol visibility

cmatzenbach opened the pull request:
I'm not sure why github thinks I added lines 31-50; I didn't, they were already there.  For everthing else, added the IUP_EXPORT annotation to all lines in the discussed range, except for the functions that were labeled as deprecated in the comments.

Commit:
SymbolVisibility: Initial changes to add symbol visibility annotations to public APIs. This is required in order to make Emscripten work because explicit export files don't really work there. Also, this will allow other platforms to control symbol visibility more easily.

cmatzenbach committed:
IIUP: Added export annotation to all func prototyes, except for
deprecated funcs

ewmailing commented:
I think the reason the extra lines are added is because this request is trying to commit to the mainline branch instead of the SymbolVisibility branch.

The mainline branch must remain clean because that is the branch I am using to track only official IUP updates (so it is possible to merge with all the other downstream branches).

I think we need to abort this request and try another on the correct branch. Alternatively, we can try to do this directly in Git. I still don't really understand the Github interface.

ewmailing commented:
I see that you actually already committed into the SymbolVisibility branch and pushed it. That kind of defeats the purpose of this GitHub UI. But you did commit to the correct branch there. So I'm going to close this.

By the way, I think you missed a few functions:



/* DEPRECATED IupTree utilities, use Iup*AttributeId functions. It will be removed in a future version.  */
void  IupTreeSetAttribute  (Ihandle* ih, const char* name, int id, const char* value);
void  IupTreeStoreAttribute(Ihandle* ih, const char* name, int id, const char* value);
char* IupTreeGetAttribute  (Ihandle* ih, const char* name, int id);
int   IupTreeGetInt        (Ihandle* ih, const char* name, int id);
float IupTreeGetFloat      (Ihandle* ih, const char* name, int id);
void  IupTreeSetfAttribute (Ihandle* ih, const char* name, int id, const char* format, ...);

/* DEPRECATED callback management. It will be removed in a future version. */
const char* IupGetActionName(void);

/* DEPRECATED font names. It will be removed in a future version.  */
char*     IupMapFont       (const char *iupfont);
char*     IupUnMapFont     (const char *driverfont);


/************************************************************************/
/*                      Pre-definided dialogs                           */
/************************************************************************/

Ihandle* IupFileDlg(void);
Ihandle* IupMessageDlg(void);
Ihandle* IupColorDlg(void);
Ihandle* IupFontDlg(void);
Ihandle* IupProgressDlg(void);

cmatzenbach committed:
SymbolVisibility: Fixed export annotations

Commit:
official: 3.21 r3989 (svn update -r3989)

r3989 is a little past the hard 3.21 release, but seems to fix a regression bug reported on the list.

Commit:
Merge branch 'official' into SymbolVisibility

* official:
  official: 3.21 r3989 (svn update -r3989)

# Conflicts:
#	include/iup.h

Commit:
SymbolVisibility: Accidental character deletion during merge

Commit:
SymbolVisibility: Discovered that IUP exports some functions that are not in the public headers. This requires more annotations in a lot of
other files.

To get started, I added a new private header, iup_export.h, which
defines IUP_EXPORTI, where the trailing I is for "internal api".
This keeps it distinct from IUP_EXPORT which can be for fully public
APIs.
Additionally, I don't want to add another public header.
And I don't want to make all these other headers include iup.h when they
do not already. So iup_export.h is the compromise.

TODO: I need to manually audit the entire export list and annotate
everything not already needed.

Commit:
SymbolVisibility: Missed the functions in iup_config.h. I duplicated the IUP_EXPORT stuff in iup_config.h and added a #ifndef IUP_EXPORT check to avoid redefinition. Maybe we need a shared header.

Commit:
official: Cherry picked iClassDlgPopup fix r4132 from 2017-05-12. The Mac modal dialog and filedlg required this fix to work correctly.

   M /trunk/iup/src/iup_class.c

improved iClassDlgPopup for more generic approach

Commit:
Merge branch 'official' into SymbolVisibility

* official:
  official: Cherry picked iClassDlgPopup fix r4132 from 2017-05-12. The Mac modal dialog and filedlg required this fix to work correctly.

Commit:
SymbolVisibility: Added iupdraw

Commit:
SymbolVisibility: Added export macros for IupWeb

Commit:
SymbolVisibility: Needed symbols for iupweb and webbrowser test

Commit:
SymbolVisibility: Added exports to iupole

Commit:
SymbolVisibility: exports needed for iupole

Commit:
SymbolVisibility: exports for iupole

Commit:
SymbolVisibility: Symbols needed for iuplua iupweblua

Commit:
SymbolVisibility: iupluaweb needs to export symbols, but the place to put it is in the header. But the use case never imports the header. Furthermore, the code generator only seems to allow one header to be specified. So the change is to put the export markers in the header, and change it to include iupluaweb.h instead of iupweb.h. Then make iupluaweb.h include iupweb.h

Commit:
SymbolVisibility: Needed for va_list functions.

Commit:
Official: revision 4693

Commit:
Merge branch 'official' into SymbolVisibility

* official:
  Official: revision 4693

# Conflicts:
#	include/iup.h
#	include/iup_config.h
#	include/iuplua.h
#	include/iupluaweb.h
#	src/gtk/iupgtk_drv.h
#	src/iup_getparam.c
#	srclua5/ctrl/il_webbrowser.c

Commit:
SymbolVisibility: iup_varg.h

Commit:
SymbolVisibility: Needed more iupdrv_* since last Iup official resync
